testing/test_git.py

```diff
@@ -39,6 +39,12 @@ def test_version_from_git(wd):
     assert wd.version.startswith('0.2')
 
 
+@pytest.mark.issue(179)
+def test_unicode_version_scheme(wd):
```

Comment:
how about turning that into a functional test passing it a few well known objects so only the `callable_or_entrypoint` function is under test?

Reply:
I thought it was important to capture the expectation that the user was supplying unicode to `version_scheme`, which in my mind is the functional requirement. I'd be fine with just a unit test on `callable_or_entrypoint` as well. Feel free to adapt it however you think is best.